### PR TITLE
add bluemix-cli (0.3.3)

### PR DESCRIPTION
**Pull request by srl295**
- [X] Checked there aren’t open [pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [X] Checked there aren’t closed [issues](https://github.com/caskroom/homebrew-cask/issues) where that cask was already refused.
- [X] When naming the cask, followed the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [X] Commit message includes cask’s name.
- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [X] `brew cask style --fix {{cask_file}}` left no offenses.
- [X] `brew cask install {{cask_file}}` worked successfully.
- [X] `brew cask uninstall {{cask_file}}` worked successfully.

### CODE REVIEW

**Comment by vitorgalvao** [comment]
What are you trying to do with `early_script` and `script`? What’s the goal?

Either way, they’re being misused. You’re looking for `uninstall_preflight`.

File: Casks/bluemix-cli.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+cask 'bluemix-cli' do
+  version '0.3.3'
+  sha256 '1ec49412f26e52cb588aa71df89e623e4e1fadd57f1fe55f99aad45ede776449'
+
+  url "https://public.dhe.ibm.com/cloud/bluemix/cli/bluemix-cli/Bluemix_CLI_#{version}.pkg"
```
Comment (vitorgalvao) [inline Casks/bluemix-cli.rb]: Needs a `# public.dhe.ibm.com/cloud/bluemix/ was verified as official when first introduced to the cask` comment above this line ([example](https://github.com/caskroom/homebrew-cask/blob/4ad02c9f087deb9e553ab0129c079d73c474add8/Casks/0ad.rb#L5); [reasoning](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment)).

**Comment by srl295** [comment]
@vitorgalvao trying to remove lines that MAY be added to either of 3 locations - `/etc/profile`, `~/.bashrc`, `~/.zshrc` - latter two may not even exist, hence separate sed invocations.

File: Casks/bluemix-cli.rb (hunk 1 above)
Comment (srl295) [inline Casks/bluemix-cli.rb]: done

**Comment by vitorgalvao** [comment]
Oh, I see now what they’re doing. Still, [`*flight` stanzas](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/flight.md) are the correct ones, there (and will be simpler to implement and more readable).

Also, to make sure no wrong lines are deleted, make `sed` use `^` and `$` for start and end on line (i.e. catch the lines that have *only and exactly that text*.

**Comment by srl295** [comment]
@vitorgalvao `sed` fixed. will look at `*flight*`. thx

**Comment by vitorgalvao** [comment]
Thank you for the contribution. It needed some fixes, so they were made in a45606ddd9292b2ca955c4c2bbe5ab583e6433e2. Your contribution is still included (and still credited to you), with the appropriate modifications. Please feel free to ask about any of the changes.

**Comment by srl295** [comment]
@vitorgalvao ok… sorry i didn't get to come back to this. but:
* sed of `/etc/profile` needs to be done as root (`sudo`)
* either or both of `~/.bashrc` or `~/.zshrc` may not exist

so uninstall isn't quite working.
I'd suggest 3 separate `system` lines, all optional (failure ignored):
1. `sudo … sed … /etc/profile`
2. `sed … ~/.bashrc`
3. `sed … ~/.zshrc`

Otherwise, uninstall fails with `sed: /etc/profile: permission denied`, and doesn't fix `~/.bashrc` (because sed has failed).

**Comment by vitorgalvao** [comment]
https://github.com/caskroom/homebrew-cask/pull/22610

**Pull request closed without merging**